plugins/module_utils/crypto/cryptography_openssh.py

```diff
@@ -0,0 +1,484 @@
+        pass
+
+    class InvalidDataError(Exception):
+        pass
+
+    class InvalidKeyFileError(Exception):
+        pass
+
+    class InvalidKeySizeError(Exception):
+        pass
+
+    class InvalidKeyTypeError(Exception):
+        pass
+
+    class InvalidPassphraseError(Exception):
+        pass
+
+    class Asymmetric_Keypair(object):
+        """Container for newly generated asymmetric keypairs or those loaded from existing files
+
+           :path: A path to an existing private key to be loaded
+           :keytype: One of rsa, dsa, ecdsa, ed25519
+           :size: The key length for newly generated keys
+           :passphrase: Secret used to encrypt or decrypt a new or existing key respectively
+        """
+
+        def __init__(self, path=None, keytype='rsa', size=None, passphrase=None):
+
+            self.__algorithm_parameters = {
+                'rsa': {
+                    'default_size': 2048,
+                    'valid_sizes': range(1024, 16384),
+                },
+                'dsa': {
+                    'default_size': 1024,
+                    'valid_sizes': [1024],
+                },
+                'ed25519': {
+                    'default_size': 256,
+                    'valid_sizes': [256],
+                },
+                'ecdsa': {
+                    'default_size': 256,
+                    'valid_sizes': [256, 384, 521],
+                    'curves': {
+                        256: ec.SECP256R1(),
+                        384: ec.SECP384R1(),
+                        521: ec.SECP521R1(),
+                    }
+                }
+            }
+
+            self.__privatekey = None
+            self.__publickey = None
+
+            if passphrase:
+                passphrase_validation_error = self.__validate_passphrase(passphrase)
+
+                if passphrase_validation_error:
+                    raise InvalidPassphraseError("%s cannot be encoded from text" % passphrase)
+
+                self.__encryption_algorithm = serialization.BestAvailableEncryption(passphrase.encode(encoding=_TEXT_ENCODING))
+            else:
+                self.__encryption_algorithm = serialization.NoEncryption()
+
+            if path:
+                self.__keytype = None
+                self.__load_keypair(path, passphrase)
+            else:
+                if keytype not in self.__algorithm_parameters.keys():
+                    raise InvalidKeyTypeError("%s is not a valid keytype. Valid keytypes are %s" % (keytype, ", ".join(self.__algorithm_parameters.keys())))
+
+                self.__keytype = keytype
+
+                if size:
+                    self.__size = size
+                else:
+                    self.__size = self.__algorithm_parameters[self.__keytype]['default_size']
+
+                self.__generate_keypair()
+
+        @property
+        def PrivateKey(self):
+            """Returns the openssh formatted private key of this key pair
```

Comment:
```suggestion
            """Returns the OpenSSH formatted private key of this key pair
```
(same in other comments)